test/lib/rules/yield-effects-spec.js

```diff
@@ -25,9 +79,73 @@ ruleTester.run("yield-effects", rule, {
   ],
   invalid: [
     {
+      code: "import { take } from 'redux-saga';"
+          + "export default function* saga() {"
+          + "  const sagaTakeEffect = take();"
+          + "  yield sagaTakeEffect;"
+          + "}",
+      errors: [ERRORS.noyield]
+    },
+    {
+      code: "import { take as sagaTake } from 'redux-saga';"
+          + "export default function* saga() {"
+          + "  const sagaTakeEffect = sagaTake();"
+          + "  yield sagaTakeEffect;"
```

Comment:
I think general function generator linting should be done in a separate plugin or eslint core.

I'll add the other tests for arrays